lib/at_cmd/at_cmd.c

```diff
@@ -440,6 +443,60 @@ int at_cmd_write(const char *const cmd,
 	return ret.code;
 }
 
+static char fmt_buf[512];
+
+int at_cmd_printf(const char *fmt, ...)
+{
+	int err;
+
+	k_mutex_lock(&mutex_fmt, K_FOREVER);
+
+	va_list args;
+	va_start(args, fmt);
+
+	err = vsnprintf(fmt_buf, sizeof(fmt_buf), fmt, args);
+	if (err < 0 || err >= sizeof(fmt_buf)) {
+		goto out;
+	}
+
+	va_end(args);
+
+	err = at_cmd_write(fmt_buf, NULL, 0, NULL);
+
+out:
+	k_mutex_unlock(&mutex_fmt);
+	return err;
+}
+
+int at_cmd_scanf(const char *cmd, const char *fmt, ...)
+{
+	int err;
+
+	k_mutex_lock(&mutex_fmt, K_FOREVER);
+
+	err = at_cmd_write(cmd, fmt_buf, sizeof(fmt_buf), NULL);
+	if (err) {
+		goto out;
+	}
+
+	va_list args;
+	va_start(args, fmt);
+
+	printk("RESP: %s\n", fmt_buf);
+	err = vsscanf(fmt_buf, fmt, args);
+	if (err == 0) {
+		err = -EBADMSG;
+	} else if (err > 0) {
+		return 0;
```

Comment:
Would it be better to keep the return value and pass it on in this case, since sometimes it is interesting to know the number of fields read?